core/tests/bank_forks.rs

```diff
@@ -386,21 +425,23 @@ mod tests {
             }
 
             let num_old_slots = num_set_roots * *add_root_interval - MAX_CACHE_ENTRIES + 1;
-            let expected_slots_to_snapshot = (num_old_slots as u64
-                ..=num_set_roots as u64 * *add_root_interval as u64)
-                .collect_vec();
+            let expected_slots_to_snapshot =
+                num_old_slots as u64..=num_set_roots as u64 * *add_root_interval as u64;
 
-            let rooted_bank = snapshot_test_config
+            let slots_to_snapshot = snapshot_test_config
                 .bank_forks
                 .get(snapshot_test_config.bank_forks.root())
-                .unwrap();
-            let slots_to_snapshot = rooted_bank.src.roots();
-            assert_eq!(slots_to_snapshot, expected_slots_to_snapshot);
+                .unwrap()
+                .src
+                .roots();
+            assert_eq!(
```

Comment:
nits: (I merged this because `nits` doesn't worth to block this useful pr. Feel free to create mini follow-up pr if this bothers you enough.) you can write `assert_eq!(..., true);` with `assert!(...);`

Reply:
Fixed in #10721